src/main/java/org/spongepowered/api/world/extent/BiomeArea.java

```diff
@@ -22,43 +22,50 @@
  * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
  * THE SOFTWARE.
  */
-
 package org.spongepowered.api.world.extent;
 
-import com.flowpowered.math.vector.Vector3i;
+import com.flowpowered.math.vector.Vector2i;
+
 import org.spongepowered.api.world.biome.BiomeType;
 
 /**
- * A volume containing biomes.
+ * A two-dimensional buffer for {@link BiomeType} data. This buffer has no
+ * direct relation to the world and changes to it are not synchronized to the
+ * world.
  */
-public interface BiomeVolume {
+public interface BiomeArea {
 
     /**
      * Get an object representing the biome at the given position.
      *
-     * <p>While {@code position} is a 3-dimensional position, biomes in
-     * Minecraft are column-based (over the X and Z plane). Therefore, the biome
-     * for all the blocks in a certain column will all be the same and changing
-     * the biome of one block in a column will change the biome for the
-     * entire column.</p>
-     *
      * @param position The position
      * @return The biome
      */
-    BiomeType getBiome(Vector3i position);
+    BiomeType getBiome(Vector2i position);
```

Comment:
I think it's very unlikely that 3D biomes will ever be required, they use up a lot of memory compared to flat. 

Using 3D coordinates here is misleading because it isn't obvious that the z coordinate is ignored, and could lead to breaks across implementations. A plugin that expects the z coordinate to be ignored and the whole column to be set being used with an implementation that has 3D biomes will be broken as the results of the calls will be vastly different.

If the need for 3D biomes does arise, we can simply add a BiomeVolume interface and implement both; the implementation of the 2D now sets columns instead of a single block.



> On Mar 2, 2015, at 4:11 PM, Simon816 <notifications@github.com> wrote:
> 
> In src/main/java/org/spongepowered/api/world/extent/BiomeArea.java:
> 
> >       * @param position The position
> >       * @return The biome
> >       */
> > -    BiomeType getBiome(Vector3i position);
> > +    BiomeType getBiome(Vector2i position);
> Using a 3 component vector allows for future-proofing the API which was the reason why it was a Vector3i
> 
> —
> Reply to this email directly or view it on GitHub.
>